presto-router/src/main/java/com/facebook/presto/router/scheduler/CustomSchedulerManager.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.router.scheduler;
+
+import com.facebook.airlift.log.Logger;
+import com.facebook.presto.spi.router.Scheduler;
+import com.facebook.presto.spi.router.SchedulerFactory;
+import com.google.common.collect.ImmutableMap;
+
+import java.io.File;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import static com.facebook.presto.util.PropertiesUtil.loadProperties;
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkState;
+import static com.google.common.base.Strings.isNullOrEmpty;
+
+public class CustomSchedulerManager
+{
+    private static final Logger log = Logger.get(CustomSchedulerManager.class);
+
+    private static final File CONFIG_FILE = new File("etc/router-config/router-scheduler.properties");
+    private static final String NAME_PROPERTY = "router-scheduler.name";
+
+    private final Map<String, SchedulerFactory> factories = new ConcurrentHashMap<>();
+    private Scheduler scheduler;
+
+    public void addSchedulerFactory(SchedulerFactory factory)
+    {
+        checkArgument(factories.putIfAbsent(factory.getName(), factory) == null,
+                "Presto Router Scheduler '%s' is already registered", factory.getName());
+    }
+
+    public void loadScheduler()
+            throws Exception
+    {
+        File configFileLocation = CONFIG_FILE.getAbsoluteFile();
+        Map<String, String> properties = new HashMap<>(loadProperties(configFileLocation));
```

Comment:
Let's surround  this in a try-catch for the IOException it can throw and not add `throws Exception` to the method signature